[Security Solution] remove unnecessary showExpandToDetails property

PhilippeOberti opened the pull request:
## Summary

Continuing the work with the aim of removing the old Security Solution alert details flyout, and replacing it by the new Security Solution expandable flyout alert details flyout (first PR [here](https://github.com/elastic/kibana/pull/167096)), this PR cleans up another AlertTable property.

The AlertTable component from the `response-ops` team provides a `showExpandToDetails` property to let the Security Solution plugin display a icon to open a flyout to show alert details. To my knowledge, this option is only used in the Security Solution plugin, in the cases page.

Using `useActionsColumn` instead of this `showExpandToDetails` allows us to use the new flyout easily, as well as showing other actions, which makes the UI more consistent with the rest of the plugin. The `showExpandDetails` property is therefore no longer necessary.

Observability UI and functionality is unchanged (we're still displaying the same actions in the actions column):

<img width="862" alt="Screenshot 2023-10-09 at 8 30 41 PM" src="https://github.com/elastic/kibana/assets/17276605/491cc408-77c6-4e95-a172-4c80e234f138">


Security Solution cases page now uses the expandable flyout and other icons in the Actions columns:

<img width="855" alt="Screenshot 2023-10-09 at 8 30 52 PM" src="https://github.com/elastic/kibana/assets/17276605/de09deda-ceaf-41f4-9bbf-a66e809f5af5">


New flyout displayed in the Security Solution cases:

https://github.com/elastic/kibana/assets/17276605/89d8afb0-91c5-41a4-8b22-002fb9e2be60

Old flyout is displayed in the Security Solution cases if the advanced settings is turned off

https://github.com/elastic/kibana/assets/17276605/36c95a2b-4b28-4bb7-8c92-377ff4ff53a8


https://github.com/elastic/security-team/issues/7461

### Checklist

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

PhilippeOberti committed:
[Security Solution] remove showExpandToDetails property and use expandable flyout on cases pages

PhilippeOberti commented:
## Files by Code Owner

### elastic/actionable-observability

* x-pack/plugins/observability/public/pages/alerts/alerts.tsx
* x-pack/plugins/observability/public/pages/overview/overview.tsx
* x-pack/plugins/observability/public/pages/rule_details/components/rule_details_tabs.tsx
* x-pack/plugins/observability/public/pages/slo_details/components/slo_detail_alerts.tsx

### elastic/apm-ui

* x-pack/plugins/apm/public/components/app/alerts_overview/index.tsx

### elastic/infra-monitoring-ui

* x-pack/plugins/infra/public/pages/metrics/hosts/components/tabs/alerts/alerts_tab_content.tsx

### elastic/kibana-localization

* x-pack/plugins/translations/translations/fr-FR.json
* x-pack/plugins/translations/translations/ja-JP.json
* x-pack/plugins/translations/translations/zh-CN.json

### elastic/response-ops

* x-pack/examples/triggers_actions_ui_example/public/components/alerts_table_sandbox.tsx
* x-pack/plugins/cases/public/components/case_view/components/case_view_alerts.test.tsx
* x-pack/plugins/cases/public/components/case_view/components/case_view_alerts.tsx
* x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_page/alerts_page.tsx
* x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table.test.tsx
* x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table.tsx
* x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.test.tsx
* x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.tsx
* x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/bulk_actions/bulk_actions.test.tsx
* x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/translations.ts
* x-pack/plugins/triggers_actions_ui/public/types.ts

### elastic/security-defend-workflows

* x-pack/plugins/security_solution/public/management/cypress/e2e/response_actions/isolate_mocked_data.cy.ts

### elastic/security-solution

* x-pack/plugins/security_solution/public/cases/pages/index.tsx
* x-pack/plugins/security_solution/public/common/components/control_columns/row_action/index.tsx
* x-pack/plugins/security_solution/public/common/lib/triggers_actions_ui/register_alerts_table_configuration.tsx
* x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx
* x-pack/plugins/security_solution/public/detections/hooks/trigger_actions_alert_table/use_actions_column.tsx
* x-pack/plugins/security_solution/public/management/cypress/e2e/response_actions/isolate_mocked_data.cy.ts
* x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/flyout/index.tsx

### elastic/security-threat-hunting-explore

* x-pack/plugins/security_solution/public/cases/pages/index.tsx

### elastic/security-threat-hunting-investigations

* x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx
* x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/flyout/index.tsx

XavierM commented on x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.test.tsx:
```diff
@@ -653,104 +651,6 @@ describe('AlertsTableState', () => {
     });
   });
 
-  describe('flyout', () => {
```
we need to do these test differently, they are still valid in my opinion.

PhilippeOberti replied:
I think we just need to test that passing something via the `useActionsColumn` is usable but we shouldn't care what type of action? Let's talk about it this week!

elasticmachine commented:
Pinging @elastic/apm-ui (Team:APM)

gergoabraham reviewed (approved):
defend workflow code changes LGTM

maryam-saeidi reviewed (approved):
AO changes LGTM!

PhilippeOberti commented:
@XavierM I had a discussion with my team and here are some notes:

First we feel that, while fixing the implementation to be correct and leverage the `useInternalFlyout` option rather than `useActionsColumn` to render the flyout, this PR might not be the best place to do it. It would increase the scope, the complexity and the number of files to review drastically. This PR bring consistency to the Security Solution codebase (though I agree not the correct way to do it, it has the merit of bring consistently not correct 😆) and removed a property that isn't used anymore...
I created [this ticket](https://github.com/elastic/security-team/issues/7872) to track this work on our side.

Second, I checked some old releases (for example `8.7` which is I believe where we implemented this new-to-us response-ops alert table) and we did not have the pagination working in the flyout (which is something that we loose by using `useActionsColumn` over `useInternalFlyout`). This means that for us, the changes in this PR aren't really breaking anything that wasn't already broken for the last 6 months +...

Third is I heard a few people talking about a potential change happening to the current response-ops alert table, which could shortly/soon be using a recent table package? If those changes are coming within the next 1-2 release(s), maybe we don't have to spend too much time fixing this here?

If you're ok with these statements, would you mind reviewing the current state of the PR? We would fix the actual implementation for everything the Threat Hunting Investigations team owns in the future!

machadoum reviewed (approved):
Explore code LGTM!

jennypavlova reviewed (approved):
Infra UI changes LGTM

PhilippeOberti committed:
bring back alerts_table_state flyout unit tests

PhilippeOberti commented:
> Let's bring back some of these tests in this file `triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.test.tsx` and I will be ok!

@XavierM the flyout unit tests are back!

XavierM commented on x-pack/test/examples/triggers_actions_ui_examples/alerts_table.ts:
```diff
@@ -107,61 +107,13 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
       );
     });
 
-    it('should open a flyout and paginate through the flyout', async () => {
```
@PhilippeOberti 
Instead of deleting this test, can we commented out and create a ticket to bring it back by creating a cell that can open the fly-out like we do in o11y. I will feel better around it.

PhilippeOberti replied:
is [this](https://github.com/elastic/kibana/pull/168048/commits/bc0b4081c9c462c1f397d269092c6863e8fe5f55) what you had in mind?

PhilippeOberti committed:
bring back one more test

XavierM reviewed (approved):
LGTM for now ;)

We still will like security solution engineer to use the `useInternalFlyout` inside of the alert table like we talked about. To avoid, your folks to re-implement a new pagination around alerts. Also, by passing this technique, we might be impacted in few release because we will start to show alerts across solution in a new alert page and we will like to use your fly-out when the alert is coming from security solution.

I am looking forward to working with the team to get back where the alert table should be.
